cpp/src/arrow/util/logging.h

```diff
@@ -62,30 +51,30 @@ namespace arrow {
 #ifdef NDEBUG
 #define ARROW_DFATAL ARROW_WARNING
 
-#define DCHECK(condition)      \
-  ARROW_IGNORE_EXPR(condition) \
-  while (false) ::arrow::internal::NullLog()
-#define DCHECK_OK(status)   \
-  ARROW_IGNORE_EXPR(status) \
-  while (false) ::arrow::internal::NullLog()
+#define DCHECK(condition)       \
+  ARROW_IGNORE_EXPR(condition); \
+  while (false) ::arrow::ArrowLogBase()
+#define DCHECK_OK(status)    \
+  ARROW_IGNORE_EXPR(status); \
+  while (false) ::arrow::ArrowLogBase()
 #define DCHECK_EQ(val1, val2) \
-  ARROW_IGNORE_EXPR(val1)     \
-  while (false) ::arrow::internal::NullLog()
+  ARROW_IGNORE_EXPR(val1);    \
+  while (false) ::arrow::ArrowLogBase()
 #define DCHECK_NE(val1, val2) \
-  ARROW_IGNORE_EXPR(val1)     \
-  while (false) ::arrow::internal::NullLog()
+  ARROW_IGNORE_EXPR(val1);    \
+  while (false) ::arrow::ArrowLogBase()
 #define DCHECK_LE(val1, val2) \
-  ARROW_IGNORE_EXPR(val1)     \
-  while (false) ::arrow::internal::NullLog()
+  ARROW_IGNORE_EXPR(val1);    \
+  while (false) ::arrow::ArrowLogBase()
 #define DCHECK_LT(val1, val2) \
-  ARROW_IGNORE_EXPR(val1)     \
-  while (false) ::arrow::internal::NullLog()
+  ARROW_IGNORE_EXPR(val1);    \
+  while (false) ::arrow::ArrowLogBase()
 #define DCHECK_GE(val1, val2) \
-  ARROW_IGNORE_EXPR(val1)     \
-  while (false) ::arrow::internal::NullLog()
+  ARROW_IGNORE_EXPR(val1);    \
+  while (false) ::arrow::ArrowLogBase()
 #define DCHECK_GT(val1, val2) \
-  ARROW_IGNORE_EXPR(val1)     \
-  while (false) ::arrow::internal::NullLog()
+  ARROW_IGNORE_EXPR(val1);    \
+  while (false) ::arrow::ArrowLogBase()
```

Comment:
Since there is `while (false)` ahead of `::arrow::ArrowLogBase()`, if `NDEBUG` is defined, this statement will not take effect (optimized out by the compiler). Is this the optimization?

Reply:
Ah yeah that will do it. Thanks